superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx

```diff
@@ -1231,7 +1231,7 @@ const DatabaseModal: FunctionComponent<DatabaseModalProps> = ({
         onClick={() => {
           setLoading(true);
           fetchAndSetDB();
-          window.location.href = '/tablemodelview/list#create';
+          window.location.href = '/dataset/add/';
```

Comment:
```suggestion
          history.push('/dataset/add/');
```

Reply:
Fixed in [`this commit`](https://github.com/apache/superset/pull/22610/commits/a72d05829371f6a375e165dd2d3739474587fa63). I found another `window.location.href` a little further down so I changed that to use `history.push()` as well.